plugins/firebase.js

```diff
@@ -134,6 +135,62 @@ export default async ( { app, store }, inject ) => {
       return;
     }
 
+    // Forces grabbing a refreshed token
+    const getFreshIdToken = async () => {
+      return await auth.currentUser.getIdToken(true);
+    };
+
+    const getIdToken = () => {
+      if (process.server) {
+        // When processing server-side, getIdToken() will only get called
+        //  during SSR, from components. This means nuxtServerInit() will have already
+        //  been called, and the auth token will have had(?) been saved.
+        //
+        // It is assumed that the token won't expire during SSR. I mean, what are the odds!
+        return store.state[VStore.$states.auth];
+      }
+      if (process.client) {
+        // Promise soup. It likely works.
+        return new Promise((resolve, reject) => {
+          const unsubscribe = listenToAuthState((user) => {
+            unsubscribe();
+            if (user) {
+              // If within 5 mins before expiration, force a token refresh
+              // Otherwise, return what you already got (idToken)
+              user.getIdToken().then( async (idToken) => {
```

Comment:
just noting that you can replace `.then()`'s with `await`'s but I'll handle that post merge.